Automattic/wp-calypso
# Modularise googleMyBusiness

This PR is one of many working on modularising state in Calypso. This one handles googleMyBusiness.

For more details on state modularisation, see the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) and p4TIVU-9lM-p2.

Note: I added all the reviewers that GitHub suggested. Please feel free to ignore the review request or pull in someone else if you're not comfortable reviewing this PR. Thank you!

Fixes #42436.

#### Changes proposed in this Pull Request

* Modularise googleMyBusiness state
* Move `ui/google-my-business` under the main state key for `googleMyBusiness`
* Tweak `googleMyBusiness` state tree and tests to accommodate the UI bits now living inside it

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/read` on the live branch or your local copy of this branch.
* Click `State` on the top right to ensure you see all state, and not just the diff for the last action: 
![image](https://user-images.githubusercontent.com/409615/73774406-d3e87d80-477b-11ea-9f59-3e42cc00101f.png)
* Analyse the tree and note that there's no `googleMyBusiness` key, even if you expand the list of keys. This indicates that the googleMyBusiness state hasn't been loaded yet.
* Click `My Sites` on the masterbar, followed by `Tools` and `Marketing` on the sidebar.
* Verify that a `googleMyBusiness` key is added with the googleMyBusiness state. This indicates that the googleMyBusiness state has now been loaded.
* Verify that googleMyBusiness-related functionality works normally. This indicates that I didn't mess up.
